ptz-pelco_p.cpp

```diff
@@ -0,0 +1,214 @@
+/* Pan Tilt Zoom camera pelco-p instance
+ *
+ * Copyright 2021 Luuk Verhagen <developer@verhagenluuk.nl>
+ *
+ * SPDX-License-Identifier: GPLv2
+ */
+
+#include "ptz-pelco_p.hpp"
+
+const PelcoPCmd STOP = PelcoPCmd(0x00, 0x00, 0x00, 0x00);
+const PelcoPCmd HOME = PelcoPCmd(0x00, 0x07, 0x00, 0x2B);
+const PelcoPCmd ZOOM_IN = PelcoPCmd(0x00, 0x20, 0x00, 0x00);
+const PelcoPCmd ZOOM_OUT = PelcoPCmd(0x00, 0x40, 0x00, 0x00);
+
+PelcoPCmd::PelcoPCmd(const unsigned char data_1, const unsigned char data_2, const unsigned char data_3, const unsigned char data_4)
+{
+	cmd[0] = data_1;
+	cmd[1] = data_2;
+	cmd[2] = data_3;
+	cmd[3] = data_4;
+
+	blog(LOG_DEBUG, "Command created %s", cmd);
+}
+
+void PTZPelcoP::attach_interface(UART* new_iface)
+{
+	if (iface)
+		iface->disconnect(this);
+	iface = new_iface;
+	if (iface)
+	{
+		connect(iface, &UART::receive, this, &PTZPelcoP::receive);
+		connect(iface, &UART::reset, this, &PTZPelcoP::reset);
+	}
+}
+
+char PTZPelcoP::checkSum(QByteArray& data)
+{
+	char sum = 0x00;
+	for (char c : data) {
+		sum ^= c;
+	}
+	return sum;
+}
+
+void PTZPelcoP::receive(const QByteArray& msg)
+{
+	QByteArray data = msg.data();
+	blog(LOG_DEBUG, "Pelco P received: %s", qPrintable(data.toHex()));
```

Comment:
I know this is just a debug stub, but the receive hook should check the address of the returned message to see if it is for this camera instance. Like VISCA, PELCO allows multiple cameras on the same serial port.